src/NumericalAlgorithms/Interpolation/InterpolationTargetWedgeSectionTorus.hpp

```diff
@@ -0,0 +1,220 @@
+#include "NumericalAlgorithms/Interpolation/SendPointsToInterpolator.hpp"
+#include "NumericalAlgorithms/Interpolation/Tags.hpp"
+#include "Parallel/ConstGlobalCache.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+namespace intrp {
+
+namespace OptionHolders {
+/// \brief A solid torus of points, useful, e.g., when measuring data from an
+/// accretion disc.
+///
+/// The torus' cross section (e.g., a cut at \f$\phi=0\f$) is a wedge-like shape
+/// bounded by \f$r_{\text{min}} \le r \le r_{\text{max}}\f$ and
+/// \f$|\theta| \le \theta_{\text{half-angle}}\f$.
+///
+/// The grid points are located on surfaces of constant \f$r\f$, \f$\theta\f$,
+/// and \f$\phi\f$. There are `NumberOfRadialPoints` points uniformly spaced in
+/// \f$r\f$ between `MinRadius` and `MaxRadius` (including these endpoints);
+/// `NumberOfThetaPoints` points uniformly spaced in \f$\theta\f$ from
+/// `ThetaHalfAngle` below the equator to `ThetaHalfAngle` above the equator,
+/// (including these endpoints); and `NumberOfPhiPoints` uniformly distributed
+/// azimuthally.
+///
+/// \note At least two grid points are required per dimension.
+///
+/// \note Input coordinates (radii, angles) are interpreted in the frame given
+/// by the `Frame` template parameter of the `Actions::WedgeSectionTorus`.
+struct WedgeSectionTorus {
+  struct MinRadius {
+    using type = double;
+    static constexpr OptionString help = {"Inner radius of torus"};
+    static type lower_bound() { return 0.0; }
+  };
+  struct MaxRadius {
+    using type = double;
+    static constexpr OptionString help = {"Outer radius of torus"};
+    static type lower_bound() { return 0.0; }
+  };
+  struct NumberOfRadialPoints {
+    using type = size_t;
+    static constexpr OptionString help = {
+        "Number of radial points, including endpoints"};
+    static type lower_bound() { return 2; }
+  };
+  struct ThetaHalfAngle {
+    using type = double;
+    static constexpr OptionString help = {
+        "Opening angle of wedge section from equator (rads)"};
+    static type lower_bound() { return 0.0; }
+    static type upper_bound() { return M_PI_2; }
+  };
+  struct NumberOfThetaPoints {
+    using type = size_t;
+    static constexpr OptionString help = {
+        "Number of theta points, including endpoints"};
+    static type lower_bound() { return 2; }
+  };
+  struct NumberOfPhiPoints {
+    using type = size_t;
+    static constexpr OptionString help = {
+        "Number of phi points, including endpoints"};
+    static type lower_bound() { return 2; }
+  };
+
+  using options =
+      tmpl::list<MinRadius, MaxRadius, NumberOfRadialPoints, ThetaHalfAngle,
+                 NumberOfThetaPoints, NumberOfPhiPoints>;
+  static constexpr OptionString help = {
+      "A torus extending from MinRadius to MaxRadius in r, ThetaHalfAngle"
+      "above and below the equator in theta, and 2pi in phi."};
+
+  WedgeSectionTorus(const double min_radius_in, const double max_radius_in,
```

Comment:
Let's not inline the constructor. Could you move it outside the class definition?

Reply:
Keep it in the header, or in a new `.cpp` file? Shall I also fix this in the previous `LineSegment` target, from which I copied the overall design?

Reply:
@nilsdeppe ping again. Do you have a preference on where I move the implementation to?

Reply:
I think a cpp file is fine (unless it's a template, then just do header). If you put it in the header and the class isn't a template you'll be back to needing to mark it `inline` :)